tests/drivers/uart/uart_async_api/testcase.yaml

```diff
@@ -27,6 +27,7 @@ tests:
     extra_configs:
       - CONFIG_USE_SEGGER_RTT=y
       - CONFIG_UART_RTT=y
+    platform_exclude: xmc45_relax_kit
```

Comment:
why we exclude xmc45_relax_kit?

Reply:
@npac-cy,

drivers.uart.async_api.rtt test started failing on xmc45_relax_kit
after f423ec255fb76641f69a65d1e57f3c6dde70b15d which set
CONFIG_SERIAL_SUPPORT_ASYNC=y if uart rtt is enabled. This conflicts
with the xmc4xxx uart driver which sets the same symbol conditionally
if the dma is enabled in the device tree. The conflict created some
warning during compilation causing the CI to fail.
   
All the uart_async tests on xmc45_relax_kit were disabled in
71d0394752f1563e9b9fcac7762c59fe49b4a399. This commit re-enables the
test on xmc45_relax_kit and only excludes the rtt test that's failing.